### DISCUSSION TITLE
SPIN in Environment

### DISCUSSION THREAD

**Discussion by andy-81** [tillsteinbach/VWsFriend#198] [Q&A]
Hi,

I have your docker image running to monitor a Golf 8 however in the app when I wish to make any changes I have to enter an SPIN. I can see in your env file it has the weconnect_pin variable however I am unable to see where this is pulled into the yml file or set in any variables for the docker image. 

Could you confirm what the environment variable should be to make use of the PIN? 

With the Golf, currently I receive the locked/unlocked state of the car in HomeKit via the image however am unable to adjust it and do not get any other options like being able to turn on the climatisations, I believe this is because the PIN is not being copied into anywhere inside the docker image.

Thanks in advance for any guidance.

**Comment by tillsteinbach** [comment]
The PIN is currently unused. The API that I use cannot lock/unlock the car yet. This may change in the summer when the MEB-Platform gets more features.
The result is that right now you can only see the status but not lock/unlock in HomeKit.
Regarding the climatization: does your car have climatisation that is controllable via the App? This is usually only the case for hybrids or BEVs.

**Comment by andy-81** [comment]
I wondered if that was the case, I tried using the cli tool and noticed it didn’t use the spin either.

the car does have a switch you can turn on to heat or cool the car but you can not set a temperature like you can on electric cars. Apparently it uses the last temp setting when you got out of the car and then works out if it needs to defrost or cool the car down.

attached is a screenshot from the app.

I have found most other functions work in your dashboard which is great as it shows fuel level, range and maps that on a graph along with recording my journeys. 

![FAAE9191-D992-43F6-9549-6A7ECFA0CE6E](https://user-images.githubusercontent.com/16067130/158765008-32724435-64f4-45ec-a102-0c6f553f9514.jpeg)

**Comment by tillsteinbach** [comment]
Is that an auxiliary heater as part of an option you selected?

If you want to have a look on all information I can get you can install [WeConnect-cli](https://github.com/tillsteinbach/WeConnect-cli). Would be interesting to see which services your car offers. Maybe I can add something when having more information.

EDIT: Just saw you already used WeConnect cli. Can you run `weconnect-cli -vv get /` would be interesting to see if I can add some of the missing attributes.

**Reply by andy-81** [reply]
The aux heater is standard on the car from what I am aware. I didn't add any option to the car when ordering (well none that were heater specific)

Here is the output from the command you asked me to run:
Vehicle: WVW
VIN:               WVW
Model:             Golf
Device Platform:   MBB_ODP
Nickname:          Golf
Role:              PRIMARY_USER
Enrollment Status: COMPLETED
User Role Status:  ENABLED
Capabilities: 51 items
        [access] disabling: False
        [automation] Status: MISSING_SERVICE disabling: False
        [charging] Status: MISSING_SERVICE disabling: False
        [climatisation] Status: MISSING_SERVICE disabling: False
        [cubicNetwork] disabling: False
        [cubicNetworkConsumption] disabling: False
        [dealerAppointment] disabling: True
        [destinationSync] Status: MISSING_SERVICE disabling: False
        [destinations] disabling: False
        [digitalKey] Status: MISSING_SERVICE disabling: False
        [emergencyCalling] disabling: False
        [fuelStatus] disabling: False
        [geofence] Status: MISSING_SERVICE disabling: False
        [googleEarth] Status: MISSING_SERVICE disabling: False
        [hybridRadio] Status: MISSING_SERVICE disabling: False
        [informationCall] Status: MISSING_SERVICE disabling: False
        [mapUpdate] Status: MISSING_SERVICE disabling: False
        [measurements] disabling: False
        [mobileDeviceKey] Status: MISSING_SERVICE disabling: False
        [news] Status: MISSING_SERVICE disabling: False
        [oilLevelStatus] disabling: False
        [onlineSpeech] Status: MISSING_SERVICE disabling: False
        [parkingInformation] Status: MISSING_SERVICE disabling: False
        [parkingPosition] disabling: False
        [roadsideAssistant] disabling: False
        [routing] Status: MISSING_SERVICE disabling: False
        [speedAlert] Status: MISSING_SERVICE disabling: False
        [state] disabling: False
        [theftWarning] disabling: False
        [transactionHistoryAntiTheftAlert] disabling: False
        [transactionHistoryAntiTheftAlertDelete] disabling: False
        [transactionHistoryDigitalKey] Status: MISSING_SERVICE disabling: False
        [transactionHistoryGeofence] Status: MISSING_SERVICE disabling: False
        [transactionHistoryGeofenceDelete] Status: MISSING_SERVICE disabling: False
        [transactionHistoryHonkFlash] disabling: False
        [transactionHistoryLockUnlock] disabling: False
        [transactionHistorySpeedAlert] Status: MISSING_SERVICE disabling: False
        [transactionHistorySpeedAlertDelete] Status: MISSING_SERVICE disabling: False
        [tripStatistics] disabling: False
        [vehicleHealthArchive] Status: MISSING_OPERATION disabling: False
        [vehicleHealthCampaigns] Status: MISSING_SERVICE disabling: False
        [vehicleHealthInspection] disabling: False
        [vehicleHealthSettings] Status: MISSING_OPERATION disabling: False
        [vehicleHealthTrigger] Status: MISSING_OPERATION disabling: False
        [vehicleHealthWakeUp] disabling: False
        [vehicleHealthWarnings] disabling: False
        [vehicleLights] disabling: False
        [vehicleWakeUp] disabling: False
        [vehicleWakeUpTrigger] disabling: False
        [weatherInformation] Status: MISSING_SERVICE disabling: False
        [webRadio] Status: MISSING_SERVICE disabling: False
Domains: 9 items
[access] Elements: 1 items
        [accessStatus] (last captured 2022-03-17T14:27:47+00:00)
                Overall Status: safe
                Doors: 6 items
                        bonnet: closed
                        frontLeft: closed
                        frontRight: closed
                        rearLeft: closed
                        rearRight: closed
                        trunk: closed
                Windows: 6 items
                        frontLeft: closed
                        frontRight: closed
                        rearLeft: closed
                        rearRight: closed
                        roofCover: unsupported
                        sunRoof: unsupported
[userCapabilities] Elements: 1 items
        [capabilitiesStatus]
                Capabilities: 50 items
                        [access] disabling: False
                        [automation] Status: MISSING_SERVICE disabling: False
                        [charging] Status: MISSING_SERVICE disabling: False
                        [climatisation] Status: MISSING_SERVICE disabling: False
                        [cubicNetwork] disabling: False
                        [cubicNetworkConsumption] disabling: False
                        [dealerAppointment] disabling: True
                        [destinationSync] Status: MISSING_SERVICE disabling: False
                        [digitalKey] Status: MISSING_SERVICE disabling: False
                        [emergencyCalling] disabling: False
                        [fuelStatus] disabling: False
                        [geofence] Status: MISSING_SERVICE disabling: False
                        [googleEarth] Status: MISSING_SERVICE disabling: False
                        [hybridRadio] Status: MISSING_SERVICE disabling: False
                        [informationCall] Status: MISSING_SERVICE disabling: False
                        [mapUpdate] Status: MISSING_SERVICE disabling: False
                        [measurements] disabling: False
                        [mobileDeviceKey] Status: MISSING_SERVICE disabling: False
                        [news] Status: MISSING_SERVICE disabling: False
                        [oilLevelStatus] disabling: False
                        [onlineSpeech] Status: MISSING_SERVICE disabling: False
                        [parkingInformation] Status: MISSING_SERVICE disabling: False
                        [parkingPosition] disabling: False
                        [roadsideAssistant] disabling: False
                        [routing] Status: MISSING_SERVICE disabling: False
                        [speedAlert] Status: MISSING_SERVICE disabling: False
                        [state] disabling: False
                        [theftWarning] disabling: False
                        [transactionHistoryAntiTheftAlert] disabling: False
                        [transactionHistoryAntiTheftAlertDelete] disabling: False
                        [transactionHistoryDigitalKey] Status: MISSING_SERVICE disabling: False
                        [transactionHistoryGeofence] Status: MISSING_SERVICE disabling: False
                        [transactionHistoryGeofenceDelete] Status: MISSING_SERVICE disabling: False
                        [transactionHistoryHonkFlash] disabling: False
                        [transactionHistoryLockUnlock] disabling: False
                        [transactionHistorySpeedAlert] Status: MISSING_SERVICE disabling: False
                        [transactionHistorySpeedAlertDelete] Status: MISSING_SERVICE disabling: False
                        [tripStatistics] disabling: False
                        [vehicleHealthArchive] Status: MISSING_OPERATION disabling: False
                        [vehicleHealthCampaigns] Status: MISSING_SERVICE disabling: False
                        [vehicleHealthInspection] disabling: False
                        [vehicleHealthSettings] Status: MISSING_OPERATION disabling: False
                        [vehicleHealthTrigger] Status: MISSING_OPERATION disabling: False
                        [vehicleHealthWakeUp] disabling: False
                        [vehicleHealthWarnings] disabling: False
                        [vehicleLights] disabling: False
                        [vehicleWakeUp] disabling: False
                        [vehicleWakeUpTrigger] disabling: False
                        [weatherInformation] Status: MISSING_SERVICE disabling: False
                        [webRadio] Status: MISSING_SERVICE disabling: False
[fuelStatus] Elements: 1 items
        [rangeStatus] (last captured 2022-03-17T14:27:47+00:00)
                Car Type: gasoline
                Total Range: 410 km
                Primary Engine: gasoline  SoC: 68% (410 km)
[vehicleLights] Elements: 1 items
        [lightsStatus] (last captured 2022-03-17T14:27:47+00:00)
                Lights: 2 items
                        right: off
                        left: off
[vehicleHealthInspection] Elements: 1 items
        [maintenanceStatus] (last captured 2022-03-17T14:27:47+00:00)
                Current milage: 319 km
                Inspection due in: 720 days
                Inspection due in: 32026 km
[vehicleHealthWarnings] Elements: 1 items
        [warningLights] (last captured 2022-03-17T14:27:47+00:00)
                Current milage: 319 km
                Warning Lights: []
[oilLevel] Elements: 1 items
        [oilLevelStatus]
                Error:  Error 4001: Internal error
                        info: Internal error, please try again later. If the problem persists, please contact our support.
                        timestamp: 2022-03-17 17:01:36+00:00
[measurements] Elements: 2 items
        [rangeStatus] (last captured 2022-03-17T14:27:47+00:00)
                Gasoline Range: 410km
        [odometerStatus] (last captured 2022-03-17T14:27:47+00:00)
                Odometer: 319km
[parking] Elements: 1 items
        [parkingPosition] (last captured 2022-03-17T14:27:47+00:00)
                Latitude: 5
                Longitude: -0


I have redacted some of the VIN and location just so I don't publish it to the world :) hopefully that has the information you require. I can not see any information in the list for enabling or disabling the aux heater, which is strange. It works in the app when I run it as the heating kicks in when I turn it on in the we connect app.

**Reply by tillsteinbach** [reply]
Interesting, it is also MBB_ODP like my 2021 Passat. I thought Golf 8 is a new platform. Does the command show any warnings? If not all items that were provided by the car are known to VWsFriend. So nothing that I could add.

**Reply by andy-81** [reply]
I will give it another go in the morning. 

No errors on the command but after running I tried turning things on in the we connect app and received errors. I have just tested it again now and everything is working. 

Would running events and then starting the auxiliary ventilation through the app provide any feedback? It looks like events opens a connection then listens for responses, is that right?

**Reply by tillsteinbach** [reply]
not exactly. It still only fetches information every few minutes. So you will not get instant feedback.

**Reply by andy-81** [reply]
I thought the golf 8 was the updated platform as well. It runs the same interface as the ID3’s and VW always talk about them as linked rather than to the Passat.

it is MIB3 platform from memory, the car only was delivered last week so not sure if it has the updated chipset which was talked about in December and they say they are rolling out to the ID3 next (Samsung chipset) not that it is any difference to the platform, just the chipsets being used to improve the responsiveness of the interface.

**Reply by tillsteinbach** [reply]
The interesting thing is then that the Golf does not share the backend with the IDs. ID platform is WCAR.
Passat is also MIB3 but the interface looks very outdated compared to the ID.

**Comment by andy-81** [comment]
Re-ran the get command and received the same results (the weconnect app is working now though)

I tried activating the heating to see if that kicked anything extra to show up but that is a no go at the minute. The heating activated in the weconnect app and showed 30 minutes till ready but didn’t show anything new in the get command.

does your Passat have the same option if it is the same platform or is that a new feature?

**Reply by tillsteinbach** [reply]
The passat is a hybrid so it has "real" preconditioning through the high voltage battery. So I guess for now I cannot make your heater work, but that does not mean it won't be possible in the future.
If you want to check you can also download the WeConnect ID app. This one is using the new API already and there you can check which features already are possible. So if you can control the heating there let me know. that would increase chances that it already works.

**Reply by andy-81** [reply]
Tried the new app and I don’t get a lot on it with the golf.

![8C4B7BC2-75CB-4C8E-8045-B5ABCB7C4653](https://user-images.githubusercontent.com/16067130/159055413-5c008b38-b772-4002-bcb4-7d18e4f9096a.jpeg)
